kioku.cabal

```diff
@@ -32,7 +32,7 @@ library
   default-extensions:  OverloadedStrings
 
   -- other-extensions:    
-  build-depends:       base >=4.8 && <4.9
+  build-depends:       base
```

Comment:
Rather omitting the bounds on `base` here, we should update them to be accurate.

Reply:
You're right that saying `base >= 4.8 && <4.13` asserts that the package works with all those versions. We then just have to decide whether we think that verifying that assertion with tests is worth it for this package and the number of versions we supposedly assert. At this point I don't see any reason to believe that is, given the scope of usage of this library.

On the other hand, removing the constraint asserts that this library works with *all* versions of `base` that have every existed, which is certainly false. I definitely think that having the constraint is better than not having it. If we want to make a change in how we deal with constraints in our "internal"-ish libraries, it would be better to put better constraints on the other dependencies.

For comparison, `orville`'s context is a bit different because it's in usage a multiple projects across multiple organizations. As such, it *is* worth the investment to verify that it builds correctly against its asserted dependencies, and the `0.9_development` branch has been set up to do that. I just don't think it's worth the same effort here, right now.

I hope that helps clarify things a bit. If not, I'm happy to get together and chat about it.

Reply:
That does clear things up.  My last question - should we set constraints on other dependencies as well?  Based on your comment, since this library isn't widely used, and not likely to be, I feel like we're fine not setting them all.

Reply:
I agree. I would continue to handle the constraints on other dependencies on an exception basis like we did with `vector-algorithms` rather than trying to hunt down the exact dependency ranges. The fact that we are targeting similar stack versions across our projects will project us enough consistency that we can get by without tracking every constraint in this library for the time being.